sds_data_manager/constructs/processing_construct.py

```diff
@@ -0,0 +1,103 @@
+
+    def __init__(
+        self,
+        scope: Construct,
+        construct_id: str,
+        vpc: ec2.Vpc,
+        volumes: Optional[list] = None,
+        **kwargs,
+    ):
+        """Set up the primary processing environment and queue.
+
+        Additional job definitions can be added to the construct later.
+
+        Parameters
+        ----------
+        scope : Construct
+            Parent construct.
+        construct_id : str
+            A unique string identifier for this construct.
+        vpc : ec2.Vpc
+            VPC into which to launch the compute instance.
+        volumes : list, optional
+            List of volumes to attach to the compute instance, by default None.
+        kwargs : dict
+            Keyword arguments.
+        """
+        super().__init__(scope, construct_id, **kwargs)
+
+        # Create compute environment
+        compute_environment = batch.FargateComputeEnvironment(
+            self,
+            "ProcessingComputeEnvironment-spot",
+            vpc=vpc,
+            vpc_subnets=ec2.SubnetSelection(subnet_type=ec2.SubnetType.PUBLIC),
+            spot=True,
+        )
+
+        # Create job queue
+        self.job_queue = batch.JobQueue(
+            self,
+            "ProcessingJobQueue",
+            compute_environments=[
+                batch.OrderedComputeEnvironment(
+                    compute_environment=compute_environment, order=1
+                )
+            ],
+        )
+
+        self.volumes = volumes
+
+    def add_job(self, job_name: str):
+        """Create an ECR repo and a job definition for the given job.
+
+        Parameters
+        ----------
+        job_name : str
+            Name of the job for which to create the job definition.
+        """
+        # Create a registry for each job definition (swe-repo)
+        container_repo = ecr.Repository(
+            self,
+            f"ECR-{job_name}",
+            repository_name=f"{job_name}-repo",
+        )
+        # Create the job definition
+        batch.EcsJobDefinition(
+            self,
+            f"ProcessingJob-{job_name}",
+            container=batch.EcsFargateContainerDefinition(
+                self,
+                f"FargateContainer-{job_name}",
+                image=ecs.ContainerImage.from_ecr_repository(
+                    repository=container_repo, tag="latest"
+                ),
+                memory=cdk.Size.mebibytes(4096),
+                cpu=1,
+                environment={"IMAP_SPICE_DIR": "/mnt/spice"},
+                volumes=self.volumes,
+                # TODO: Do we need to explicitly specify architecture and OS family?
+                #       We are building containers in GitHub Actions and need to
+                #       make sure these are aligned.
+                # fargate_cpu_architecture=ecs.CpuArchitecture.ARM64,
+                # fargate_operating_system_family=ecs.OperatingSystemFamily.LINUX
+            ),
+        )
```

Comment:
Here, we are missing batch job role as well. I am pretty sure it needs it as well.

Reply:
This works in my dev environment. What role/action do you think we're missing?

Reply:
Like these permission for the container:
- service-role/AmazonECSTaskExecutionRolePolicy
- permission to pull images from ECR

Like these permission for the batch role:
- iam.ManagedPolicy.from_aws_managed_policy_name("AmazonS3FullAccess"),
- iam.ManagedPolicy.from_aws_managed_policy_name(
                    "AmazonElasticFileSystemFullAccess"
                ),

Reply:
We discussed why those permission are not needed on the call.